### PR TITLE
refactor(linter): add missing `should_run` implementations

### PR DESCRIPTION
**Pull request by camchenry**
Added some `should_run` implementations that probably should exist but didn't previously.

### CODE REVIEW

**Comment by camchenry** [comment]
* **#6666** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6666?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/typescript/no_namespace.rs [hunk 1]
```diff
@@ -104,6 +100,9 @@ impl Rule for NoNamespace {
     }
 
     fn should_run(&self, ctx: &ContextHost) -> bool {
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/typescript/no_namespace.rs:102]: Looks like a bug? We're returning false when definition files are allowed and the current file is a definition file?

Comment (camchenry) [reply crates/oxc_linter/src/rules/typescript/no_namespace.rs:102]: I think it's correct: if we're looking at a - definition file and definition files are allowed, then we don't need to emit any diagnostics do should_run be false. Does that seem incorrect?

Comment (DonIsaac) [reply crates/oxc_linter/src/rules/typescript/no_namespace.rs:102]: I think I misunderstood the meaning of "allow definition files". I believe you're correct

**Comment by DonIsaac** [comment]
### Merge activity

* **Oct 19, 1:53 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**